infrastructure/balsa_queue/gonogo/gonogo.hh

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <string>
+
+namespace jet {
+
+class GoNoGo {
+  public: 
+   GoNoGo();
+   void go(std::string message);
+   void nogo(std::string message);
+   bool isReady();
+   std::string getMessage();
+
+  private: 
+   bool ready;
+   std::string message;
```

Comment:
If you agree that we don't need to store a string while in a `go()` state, you can replace this `std::string` with a `std::optional<std::string>`. When a GoNoGo message is published with `ready = true`, the message on the wire won't include any overhead from the empty message string.